### PR TITLE
Db setup for ecommerce

### PR DESCRIPTION
**Pull request by imran-akhtar-brainx**

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Category < ApplicationRecord
+  has_many :sub_categories
```
Comment (usama-javaid-brainx) [inline app/models/category.rb]: use dependent destroy here as well

File: .idea/workspace.xml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      <change beforePath="$PROJECT_DIR$/config/routes.rb" beforeDir="false" afterPath="$PROJECT_DIR$/config/routes.rb" afterDir="false" />
+    </list>
+    <option name="SHOW_DIALOG" value="false" />
+    <option name="HIGHLIGHT_CONFLICTS" value="true" />
+    <option name="HIGHLIGHT_NON_ACTIVE_CHANGELIST" value="false" />
+    <option name="LAST_RESOLUTION" value="IGNORE" />
+  </component>
+  <component name="Git.Settings">
+    <option name="RECENT_GIT_ROOT_PATH" value="$PROJECT_DIR$" />
+  </component>
```
Comment (usama-javaid-brainx) [inline .idea/workspace.xml]: remove all idea files as these are the IDE configurations

File: db/migrate/20221125065738_create_products.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateProducts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :products do |t|
+      t.belongs_to :sub_category
```
Comment (usama-javaid-brainx) [inline db/migrate/20221125065738_create_products.rb]: you didn't mention foreign_key here

File: app/models/sub_category.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class SubCategory < ApplicationRecord
+  belongs_to :category
+  has_many :products
```
Comment (usama-javaid-brainx) [inline app/models/sub_category.rb]: dependant destroy here as well

File: db/migrate/20221125070346_create_orders.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class CreateOrders < ActiveRecord::Migration[7.0]
+  def change
+    create_table :orders do |t|
+      t.string :email
+      t.string :customer_name
+      t.integer :phone_number
```
Comment (usama-javaid-brainx) [inline db/migrate/20221125070346_create_orders.rb]: phone number must be string

File: test/fixtures/categories.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+# This model initially had no columns defined. If you add columns to the
+# model remove the "{}" from the fixture names and add the columns immediately
+# below each fixture, per the syntax in the comments below
+#
+one: {}
+# column: value
+#
+two: {}
+# column: value
```
Comment (usama-javaid-brainx) [inline test/fixtures/categories.yml]: remove all the test files as we are not using it

**Review by usama-javaid-brainx — CHANGES_REQUESTED** [review]

File: db/seeds.rb [hunk 7]
```diff
@@ -5,3 +5,8 @@
 #
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
+Category.create(name: "Men's")
+Category.create(name: "women's")
+Product.create(sub_category_id:1, name: "Nike Retro", description: "Created for the hardwood but taken verlays and original.", price: 120.5)
+SubCategory.create(category_id: 1, name: "Shoes", description: "80s b-ball icon returns with perfectly")
+Order.create(email: "imran.akhtar@gmail.com", customer_name: "imran", phone_number: 03340053857)
```
Comment (sobanakram) [inline db/seeds.rb]: Use variables and use id through variable instead of static id 
Also, create around 5 sub categories and 10 products here.

**Review by sobanakram — APPROVED** [review]

**Pull request merged**